File: homestay-backend/Homestay.Api/Domain/Entities/Payment.cs
using System;
using System.Collections.Generic;

namespace Homestay.Api.Domain.Entities;

public partial class Payment
{
    public Guid Id { get;  set; }

    public Guid BookingId { get;  set; }

    public decimal Amount { get; set; }

    public string Currency { get; set; } = null!;

    public string PaymentMethod { get; set; } = null!;

    public string Status { get; set; } = null!;

    public DateTime? PaidAt { get; set; }

    public DateTime? CreatedAt { get; private set; }

    public virtual Booking Booking { get; private set; } = null!;
}

File: homestay-frontend/types/payment.ts
export type PaymentStatus = "PENDING" | "SUCCESS" | "FAILED";

export interface Payment {
  id: string;
  bookingId: string;
  amount: number;
  currency: string;
  status: PaymentStatus;
  paymentMethod?: string;
  transactionId?: string;
  createdAt: string;
  paidAt?: string;
}

export interface PaymentResponse {
  id: string;
  bookingId: string;
  amount: number;
  currency: string;
  status: PaymentStatus;
  paymentMethod?: string;
  transactionId?: string;
  createdAt: string;
  paidAt?: string;
}
